app/main/views/suppliers.py

```diff
@@ -439,6 +440,57 @@ def find_supplier_services():
     )
 
 
+@main.route(
+            '/suppliers/<string:supplier_name>/<int:supplier_code>/<string:domain_name>/'
+            '<int:supplier_domain_id>/<string:price>/update',
+            methods=['POST', "GET"])
+@login_required
+@role_required('admin')
+def update_supplier_category_price(supplier_name, supplier_code, domain_name, supplier_domain_id, price):
+    errorMsg = None
+    if request.args.get('form') is not None:
+        errorMsg = 'Please enter a valid price'
+
+    form = SellerUpdateCategoryPriceForm(request.form)
+    return render_template_with_csrf(
+           'update_category_price.html', form=form, supplier_name=supplier_name, supplier_code=supplier_code,
+           supplier_domain_id=supplier_domain_id, domain_name=domain_name, price=price, errorMsg=errorMsg)
+
+
+@main.route(
+            '/suppliers/<int:supplier_code>/<supplier_domain_id>/<string:domain_name>/<string:supplier_name>/update',
+            methods=['POST'])
+@login_required
+@role_required('admin')
+def update_price(supplier_code, supplier_domain_id, domain_name, supplier_name):
+    form = SellerUpdateCategoryPriceForm(request.form)
+
+    if not isPriceValid(request.form['new_supplier_category_price']):
```

Comment:
Have a look at other examples of `form.validate()`.  I think we can follow the same pattern here.

Reply:
this part is no longer in module